x-pack/plugins/maps/public/connected_components/timeslider/timeslider.tsx

```diff
@@ -0,0 +1,179 @@
+    this.setState({ range: (value[1] as number) - (value[0] as number) }, () => {
+      this._onChange(value as [number, number]);
+    });
+  };
+
+  _onChange = (value: [number, number]) => {
+    this.setState({
+      timeslice: value,
+      timesliceText: prettyPrintTimeslice(value),
+    });
+    this._propagateChange(value);
+  };
+
+  _onNext = () => {
+    const from =
+      this._doesTimesliceCoverTimerange() || this.state.timeslice[1] === this.state.max
+        ? this.state.ticks[0].value
+        : this.state.timeslice[1];
+    const to = from + this.state.range;
+    this._onChange([from, to <= this.state.max ? to : this.state.max]);
+  };
+
+  _onPrevious = () => {
+    const to =
+      this._doesTimesliceCoverTimerange() || this.state.timeslice[0] === this.state.min
+        ? this.state.ticks[this.state.ticks.length - 1].value
+        : this.state.timeslice[0];
+    const from = to - this.state.range;
+    this._onChange([from < this.state.min ? this.state.min : from, to]);
+  };
+
+  _propagateChange = _.debounce((value: [number, number]) => {
+    if (this._isMounted) {
+      this.props.setTimeslice({ from: value[0], to: value[1] });
+    }
+  }, 300);
+
+  render() {
+    return (
+      <div className="mapTimeslider">
+        <div className="mapTimeslider__row">
+          <EuiButtonIcon
+            onClick={this.props.closeTimeslider}
+            iconType="cross"
+            color="subdued"
+            className="mapTimeslider__close"
+            aria-label={i18n.translate('xpack.maps.timeslider.closeLabel', {
+              defaultMessage: 'Close timeslider',
+            })}
+          />
+
+          <div className="mapTimeslider__timeWindow">
+            <EuiText size="s">{this.state.timesliceText}</EuiText>
+          </div>
+
+          <div className="mapTimeslider__innerPanel">
+            <div className="mapTimeslider__controls">
+              <EuiButtonIcon
+                onClick={this._onPrevious}
+                iconType={PreviousTimesliceIcon}
+                color="text"
+                aria-label={i18n.translate('xpack.maps.timeslider.previousTimeWindowLabel', {
+                  defaultMessage: 'Previous time window',
+                })}
+              />
+              <EuiButtonIcon
+                onClick={this._onNext}
+                iconType={NextTimesliceIcon}
+                color="text"
+                aria-label={i18n.translate('xpack.maps.timeslider.nextTimeWindowLabel', {
+                  defaultMessage: 'Next time window',
+                })}
+              />
+            </div>
+          </div>
+        </div>
+
+        <div className="mapTimeslider__row">
+          <EuiDualRange
+            fullWidth={true}
+            value={this.state.timeslice}
+            onChange={this._onDualControlChange}
+            showTicks={true}
+            min={this.state.min}
+            max={this.state.max}
+            step={1}
```

Comment:
Instead of `step={1}` shouldn't we be using instead `step={this.state.range}`? So that users could only move the thumbs to the ticks? 

If they can move the thumbs to values in between ticks, this can happen:

https://user-images.githubusercontent.com/2750668/117326428-8af50c80-ae89-11eb-8c6d-f434cf9e60fd.mov



@nreese and @thompsongl my assumption is that the step should be `step={this.state.range}` but when I try that the code fails with: 

```
Uncaught Error: The value of 1619740800000 is not included in the possible
sequence provided by the step of 86400000 
at EuiRangeTrack.validateValueIsInStep (range_track.js:84)
```

Reply:
I'll leave it up to the Maps folks to decide how/if they want to limit selected ranges, but I have been working through step locking when it comes to the draggable highlight.
Especially when the step ticks are as far apart as they are in that video, it's very easy to get into interactions that feel off. That is, a small drag either results in no movement or movement that feels extreme and out of sync with the cursor.
Still working on it, but I think it'll be debounced so that the step only increments when you've dragged 50% of the step range.

Reply:
> shouldn't we be using instead step={this.state.range}? So that users could only move the thumbs to the ticks?

I don't think so. The ticks are not very granular. Users will want very precise control over the selected time slice, down to minute or second or even millisecond depending on the use case. I have seen analysts working with time slider tools like these in past jobs and the time slice they want is often based on the data. They may want to narrow the time slice to show just a single document and flip back and forth between steps to play the change over and over.